gateway/tenant_test.go

```diff
@@ -0,0 +1,72 @@
+package gateway
+
+import (
+	"errors"
+	"os"
+	"reflect"
+	"testing"
+)
+
+func TestGetTenants(t *testing.T) {
+	testCases := []struct {
+		name            string
+		filePath        string
+		expectedTenants Tenant
+		expectedErr     error
+	}{
+		{
+			name:     "Valid input file",
+			filePath: "testdata/valid.yaml",
+			expectedTenants: Tenant{
+				All: map[string]tenant{
+					"username1": {
+						ID:       "orgid1",
+						Username: "username1",
+						Password: "password1",
+					},
+					"username2": {
+						ID:       "orgid2",
+						Username: "username2",
+						Password: "password2",
+					},
+				},
+			},
+			expectedErr: nil,
+		},
+		{
+			name:            "Invalid input file",
+			filePath:        "testdata/invalid.yaml",
+			expectedTenants: Tenant{},
+			expectedErr:     errors.New("yaml: unmarshal errors:\n  line 2: cannot unmarshal !!seq into map[string]gateway.tenant"),
```

Comment:
very specific. if it becomes too hard to keep it working, you can just verify err != nil or match error string with a specific string. No need to change it if it is working without trouble 👍